File: src/main/java/io/realworld/api/response/Comment.java
package io.realworld.api.response;

import java.time.Instant;

public class Comment {
    private Long id;
    private String body;
    private Instant createdAt;
    private Instant updatedAt;
    private Profile author;

    public Long getId() {
        return id;
    }

    public void setId(final Long id) {
        this.id = id;
    }

    public String getBody() {
        return body;
    }

    public void setBody(final String body) {
        this.body = body;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(final Instant createdAt) {
        this.createdAt = createdAt;
    }

    public Instant getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(final Instant updatedAt) {
        this.updatedAt = updatedAt;
    }

    public Profile getAuthor() {
        return author;
    }

    public void setAuthor(final Profile author) {
        this.author = author;
    }
}

File: backend/src/main/kotlin/com/hexagonkt/realworld/services/Comment.kt
package com.hexagonkt.realworld.services

import com.hexagonkt.core.fieldsMapOfNotNull
import java.time.LocalDateTime

data class Comment(
    val id: Int,
    val author: String,
    val body: String,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val updatedAt: LocalDateTime = LocalDateTime.now()
) {
    fun toMap(): Map<String, *> =
        fieldsMapOfNotNull(
            Comment::id to id,
            Comment::author to author,
            Comment::body to body,
            Comment::createdAt to createdAt,
            Comment::updatedAt to updatedAt,
        )
}
